File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        List<Integer> list=new ArrayList<>();
        ListNode temp=head;
        while(temp!=null){
            list.add(temp.val);
            temp=temp.next;
        }
        if(list.size()>=n){
            list.remove(list.size()-n);
        }
        ListNode dummy=new ListNode(0);
        ListNode t=dummy;
        for(int l:list){
            t.next=new ListNode(l);
            t=t.next;
        }
        return dummy.next;
    }
}

File: 19-remove-nth-node-from-end-of-list/remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: Optional[ListNode]
        :type n: int
        :rtype: Optional[ListNode]
        """
        s=head
        f=head
        for i in range(n):
            f=f.next
        if f==None:
            return head.next
        while f and f.next:
            s=s.next
            f=f.next
        s.next=s.next.next
        return head
        